src/control/lib/control/request.go

```diff
@@ -80,12 +98,45 @@ func (r *request) AddHost(hostAddr string) {
 	r.HostList = append(r.HostList, hostAddr)
 }
 
+// SetTimeout sets a deadline by which the request must have
+// completed. It is calcluated from the supplied time.Duration.
+func (r *request) SetTimeout(timeout time.Duration) {
+	r.deadline = time.Now().Add(timeout)
+}
+
+// getDeadline retrieves the deadline set for the request, if any.
+// Callers should check the returned time.Time for the zero value.
+func (r *request) getDeadline() time.Time {
+	return r.deadline
+}
+
 // isMSRequest implements part of the targetChooser interface,
 // and will always return false for a basic request.
 func (r *request) isMSRequest() bool {
 	return false
 }
 
+// canRetry implements the retryer interface and always
+// returns false, indicating that the default request implementation
+// is not retryable.
+func (r *request) canRetry(_ error, _ uint) bool {
+	return false
+}
+
+// onRetry implements the retryer interface and always returns
+// an error. Callers should check the result of canRetry() before
+// calling retry, in order to avoid wasting effort on a request
+// that does not implement its own retry logic.
```

Comment:
This part of the comment seems to be more about the interface in general than about this specific implementation. I think it'd be helpful to put any general comments about how the interface aught to be used near the interface definition.

Reply:
fixed